src/coreComponents/physicsSolvers/fluidFlow/wells/SinglePhaseWell.cpp

```diff
@@ -106,14 +108,20 @@ void SinglePhaseWell::validateWellConstraints( MeshLevel const & meshLevel ) con
   {
     WellControls const & wellControls = getWellControls( subRegion );
     WellControls::Control const currentControl = wellControls.getControl();
-    real64 const targetTotalRate = wellControls.getTargetTotalRate();
-    real64 const targetPhaseRate = wellControls.getTargetPhaseRate();
-    GEOSX_ERROR_IF( currentControl == WellControls::Control::PHASEVOLRATE,
-                    "Phase rate control is not available for SinglePhaseWell" );
-    GEOSX_ERROR_IF( isZero( targetTotalRate ),
-                    "Target total rate cannot be equal to zero" );
-    GEOSX_ERROR_IF( !isZero( targetPhaseRate ),
-                    "Target phase rate cannot be used for SinglePhaseWell" );
+    real64 const targetTotalRate = wellControls.getTargetTotalRate( m_currentTime ); // validation done at t = 0
+    real64 const targetPhaseRate = wellControls.getTargetPhaseRate( m_currentTime ); // validation done at t = 0
+    GEOSX_THROW_IF( currentControl == WellControls::Control::PHASEVOLRATE,
+                    "WellControls named " << wellControls.getName() <<
+                    ": Phase rate control is not available for SinglePhaseWell",
+                    InputError );
+    GEOSX_THROW_IF( isZero( targetTotalRate ),
+                    "WellControls named " << wellControls.getName() <<
+                    ": Target total rate cannot be equal to zero",
+                    InputError );
```

Comment:
Just to make it clear, does this mean one cannot start the simulation with the well shut-in? Is it because of some numerical problem/limitation?

Reply:
It was a mistake in fact. I removed this check and modified an integrated test to start with a zero rate (i.e. shut down)